src/state/slices/txHistorySlice/txHistorySlice.ts

```diff
@@ -138,34 +138,39 @@ const updateOrInsertTxs = (txHistory: TxHistory, incomingTxs: Tx[], accountId: A
 export const txHistory = createSlice({
   name: 'txHistory',
   initialState,
-  reducers: {
-    clear: () => {
+  reducers: create => ({
+    clear: create.reducer(() => {
       return initialState
-    },
-    onMessage: (txState, { payload }: TxMessage) => {
+    }),
+    onMessage: create.reducer((txState, { payload }: TxMessage) => {
       updateOrInsertTxs(txState, [payload.message], payload.accountId)
-    },
-    upsertTxsByAccountId: (txState, { payload }: TxsMessage) => {
+    }),
+    upsertTxsByAccountId: create.reducer((txState, { payload }: TxsMessage) => {
       for (const [accountId, txs] of Object.entries(payload)) {
         updateOrInsertTxs(txState, txs, accountId)
       }
 
       return txState
-    },
-    setAccountIdHydrated: (txState, { payload }: { payload: AccountId }) => {
+    }),
+    setAccountIdHydrated: create.reducer((txState, { payload }: { payload: AccountId }) => {
       txState.hydrationMeta[payload] = {
         minTxBlockTime: txState.hydrationMeta[payload]?.minTxBlockTime,
         isHydrated: true,
         isErrored: false,
       }
-    },
-    setAccountIdErrored: (txState, { payload }: { payload: AccountId }) => {
+    }),
+    setAccountIdErrored: create.reducer((txState, { payload }: { payload: AccountId }) => {
       txState.hydrationMeta[payload] = {
         minTxBlockTime: txState.hydrationMeta[payload]?.minTxBlockTime,
         isHydrated: false,
         isErrored: true,
       }
-    },
+    }),
+  }),
+  selectors: {
+    selectTxsById: state => state.txs.byId,
+    selectTxIds: state => state.txs.ids,
```

Comment:
can add `selectTxIdsByAccountIdAssetId` and replace `selectors:L100`

Reply:
6634065421

Reply:
couple more:
- add `selectEnabledAccountIds` and replace `common-selectors:L38`
- use `selectAccountIdsByWalletId` for `common-selectors:L47`
- add `selectAccountBalancesById` and replace `common-selectors:L83`

Reply:
4f8708dfda90456c1ad28e297bb5b20920d65d23